match/lib/match/git_helper.rb

```diff
@@ -79,6 +81,42 @@ def self.clear_changes
       @dir = nil
     end
 
+    # Create and checkout an specific branch in the git repo
+    def self.checkout_branch(branch)
+      return unless @dir
+
+      commands = []
+      if branch_exists?(branch)
+        commands << "git checkout #{branch}"
+      else
+        commands << "git checkout --orphan #{branch}"
+        commands << "git reset --hard"
+      end
+
+      UI.message "Checking out branch #{branch}..."
+
+      Dir.chdir(@dir) do
+        commands.each do |command|
+          FastlaneCore::CommandExecutor.execute(command: command,
+                                                print_all: $verbose,
+                                                print_command: $verbose)
+        end
+      end
+    end
+
+    # Checks if a specific branch exists in the git repo
+    def self.branch_exists?(branch)
+      return unless @dir
+
+      result = nil
+      Dir.chdir(@dir) do
+        result = FastlaneCore::CommandExecutor.execute(command: "git branch --list #{branch} --no-color",
+                                                       print_all: $verbose,
+                                                       print_command: $verbose)
+      end
```

Comment:
Because `Dir.chdir` will return the value returned by the block as its own result, you can shorten this slightly to:

```ruby
result = Dir.chdir(@dir) do
  FastlaneCore::CommandExecutor.execute(command: "git branch --list #{branch} --no-color",
                                      print_all: $verbose,
                                  print_command: $verbose)
end
```